### PR TITLE
Support for Async SSL connection parameters

### PR DESCRIPTION
**Pull request by hverr**
This pull requests adds support for Async SSL connection parameters, e.g. a CA certificate file.

The pull request maintains a backwards compatible interface and allows compilation without support for SSL.

### CODE REVIEW

**Comment by prepor** [comment]
This will be merged?

**Comment by little-arhat** [comment]
Cool! This is extremely useful nowadays, since many services, i.e. docker, want to use TLS and its self-signed certificates! Looking forward to merging this into ocaml-conduit.

**Comment by prepor** [comment]
Yes, I need it in my https://github.com/prepor/condo/ to communicate with docker :)

**Comment by hverr** [comment]
Is there any follow-up on this? Any chance this will be merged?

**Comment by rgrinberg** [comment]
@hverr I'll give it a look tomorrow. Sorry for the delay.

File: lib/conduit_async.ml [hunk 1]
```diff
@@ -22,12 +22,44 @@ IFDEF HAVE_ASYNC_SSL THEN
...
+  let configure ?version ?name ?ca_file ?ca_path ?session ?verify () =
+    { version; name; ca_file; ca_path; session; verify}
+ELSE
+  type config = unit with sexp
+
+  let verify_certificate connection =
+    let _ = connection in return true
+
+  let configure ?version ?name ?ca_file ?ca_path ?session ?verify () =
+    let _, _, _, _, _, _ = (version, name, ca_file, ca_path, session, verify) in
```
Comment (rgrinberg) [inline lib/conduit_async.ml]: Wouldn't it be better if `configure` and `verify_certificate` just raised in this clause of the ifdef?

**Comment by rgrinberg** [comment]
@hverr sorry for the late reply, just one simple question and I think this is fine.

**Comment by hverr** [comment]
@rgrinberg No problem, I'm sure you were busy doing other things. I have changed the code to always raise an exception when SSL is not available, as requested.

**Pull request merged**